com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -0,0 +1,186 @@
+            }
+        }
+        return withoutContent;
+    }
+
+    @Override
+    public void deleteFile(String id, String space) throws FileStorageException {
+        blobStore.removeBlob(container, id);
+    }
+
+    @Override
+    public void deleteFilesBySpace(String space) throws FileStorageException {
+        removeBlobsByFilter(blob -> filterBySpace(blob, space));
+    }
+
+    @Override
+    public void deleteFilesBySpaceAndNamespace(String space, String namespace) throws FileStorageException {
+        removeBlobsByFilter(blob -> filterBySpaceAndNamespace(blob, space, namespace));
+    }
+
+    @Override
+    public int deleteFilesModifiedBefore(Date modificationTime) throws FileStorageException {
+        return removeBlobsByFilter(blob -> filterByModificationTime(blob, modificationTime));
+    }
+
+    @Override
+    public void processFileContent(FileDownloadProcessor fileDownloadProcessor) throws FileStorageException {
+        FileEntry fileEntry = fileDownloadProcessor.getFileEntry();
+        InputStream fileContentStream = null;
+        try {
+            Blob blob = blobStore.getBlob(container, fileEntry.getId());
+            Payload payload = blob.getPayload();
+            fileContentStream = payload.openStream();
+            fileDownloadProcessor.processContent(fileContentStream);
+        } catch (Exception e) {
+            throw new FileStorageException(e);
+        } finally {
+            if (fileContentStream != null) {
+                closeQuietly(fileContentStream);
+            }
+        }
+    }
+
+    private Map<String, String> createFileEntryMetadata(FileEntry fileEntry) {
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put(FileService.FileServiceColumnNames.SPACE.toLowerCase(), fileEntry.getSpace());
+        metadata.put(FileService.FileServiceColumnNames.FILE_NAME.toLowerCase(), fileEntry.getName());
+        metadata.put(FileService.FileServiceColumnNames.MODIFIED.toLowerCase(), Long.toString(fileEntry.getModified()
+            .getTime()));
+        if (fileEntry.getNamespace() != null) {
+            metadata.put(FileService.FileServiceColumnNames.NAMESPACE.toLowerCase(), fileEntry.getNamespace());
+        }
+        return metadata;
+    }
+
+    private int removeBlobsByFilter(Predicate<? super StorageMetadata> filter) {
+        Set<String> entriesToDelete = blobStore.list(container, new ListContainerOptions().withDetails())
+            .stream()
+            .filter(filter)
+            .map(blob -> blob.getName())
+            .collect(Collectors.toSet());
+
+        if (entriesToDelete.size() > 0) {
+            blobStore.removeBlobs(container, entriesToDelete);
+        }
+        return entriesToDelete.size();
+    }
+
+    private boolean filterByModificationTime(StorageMetadata blobMetadata, Date modificationTime) {
+        Map<String, String> userMetadata = blobMetadata.getUserMetadata();
+        if (userMetadata == null || userMetadata.isEmpty()) {

```

Comment:
Not very clear why we are checking the userMetadata and why we return true in case it is missing?
Also, could you use CollectionUtils.isNullOrEmpty ?

Reply:
Done